### PR TITLE
Array.test.ts

### PR DESCRIPTION
**Pull request by gcrev93**
Unit test for array helper function

Also have a similar request as seen in PR #175  to have the array functions throw errors if value isnt found. One function naturally throws an error and the other does not. 

If approved will change functions and test accordingly

### CODE REVIEW

File: src/tests/unit/helpers/array.test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            slotName: 'slot4',
+            abilityName: 'ability2'
+        }
+
+        const actual = arrayHelpers.removeSlotFromSlotIdArray(slotIds, singleSlotId)
+        expect(actual).toEqual(slotIds)
+    })
+
+    test('when SlotID does exist', () => {
+        const abilities: Ability<MockUserState, AllSyncStorageLayer<MockUserState>>[] = [{
```
Comment (howlowck) [inline src/tests/unit/helpers/array.test.ts]: You don't need `abilities`

File: src/tests/unit/helpers/array.test.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            [{ slotName: 'slot1', abilityName: 'ability2' },
+            { slotName: 'slot3', abilityName: 'ability1' }]
+        )
+
+    })
+})
+
+describe('removeSlotFromSlotIdArray function', () => {
+    test('when SlotID does not exist', () => {
+        const abilities: Ability<MockUserState, AllSyncStorageLayer<MockUserState>>[] = [{
```
Comment (howlowck) [inline src/tests/unit/helpers/array.test.ts]: You don't need the `abilities` array here to run the test.

File: src/tests/unit/helpers/array.test.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        ]
+
+        expect(() => arrayHelpers.changeArrayItemOnIndex(slotIds, 3, { slotName: slotIds[3].slotName, abilityName: 'ability1' })).toThrowError()
+        //console.log(actual)
+        //expect(actual).toEqual(-1)
+        //expect(() => findHelpers.findIndexOfSlotIdsBySlotId(slotIds, singleSlotId)).toThrow()
+    })
+
+    test('when SlotID does exist', () => {
+        const abilities: Ability<MockUserState, AllSyncStorageLayer<MockUserState>>[] = [{
```
Comment (howlowck) [inline src/tests/unit/helpers/array.test.ts]: Same as the other comments regarding `abilities`.

File: src/tests/unit/helpers/array.test.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            abilityName: 'ability1'
+        },
+        {
+            slotName: 'slot3',
+            abilityName: 'ability1'
+        },
+        ]
+
+        expect(() => arrayHelpers.changeArrayItemOnIndex(slotIds, 3, { slotName: slotIds[3].slotName, abilityName: 'ability1' })).toThrowError()
+        //console.log(actual)
```
Comment (howlowck) [inline src/tests/unit/helpers/array.test.ts]: Remove the comments

File: src/tests/unit/helpers/array.test.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+interface MockUserState {
+    firstName: string,
+    lastName: SVGAnimatedString,
+    age: number
+}
+
+describe('changeArrayItemOnIndex function', () => {
+    test('when SlotID does not exist', () => {
+        const abilities: Ability<MockUserState, AllSyncStorageLayer<MockUserState>>[] = [{
```
Comment (howlowck) [inline src/tests/unit/helpers/array.test.ts]: You don't need `abilities`

**Review by howlowck — CHANGES_REQUESTED** [review]
Remove all the `abilities` objects that are not used in the tests.  (this means you probably don't need `MockUserState` either

File: src/tests/unit/helpers/array.test.ts (hunk 1 above)
Comment (gcrev93) [reply src/tests/unit/helpers/array.test.ts]: @howlowck removed abilities and MockUserState

**Review by howlowck — APPROVED** [review]

**Comment by howlowck** [comment]
nice!

**Pull request merged**